kms/client-examples_test.go

```diff
@@ -264,3 +267,42 @@ func ExampleClient_ListEnclaves() {
 		fmt.Println(v.Name)
 	}
 }
+
+// ExampleClient_Logs shows how to fetch server log records.
+func ExampleClient_Logs() {
+	key, err := kms.ParseAPIKey("k1:d7cY_5k8HbBGkZpoy2hGmvkxg83QDBXsA_nFXDfTk2E")
+	if err != nil {
+		log.Fatalf("Failed to parse KMS API key: %v", err)
+	}
+
+	client, err := kms.NewClient(&kms.Config{
+		Endpoints: []string{
+			"127.0.0.1:7373",
+		},
+		APIKey: key,
+		TLS: &tls.Config{
+			RootCAs:            nil,   // Use nil for system root CAs or customize
+			InsecureSkipVerify: false, // Don't skip TLS cert verification in prod
+		},
+	})
+	if err != nil {
+		log.Fatalf("Failed to create KMS client: %v", err)
+	}
+
+	logs, err := client.Logs(context.TODO(), &kms.LogRequest{
+		Host:  "127.0.0.1:7373",                 // The server to fetch logs from
+		Level: slog.LevelWarn,                   // Fetch only warnings or error logs
+		Since: time.Now().Add(-5 * time.Minute), // Fetch logs of the last 5 min
+	})
+	if err != nil {
+		log.Fatalf("Failed to fetch server logs: %v", err)
+	}
+	defer logs.Close()
+
+	for r, ok := logs.Next(); ok; r, ok = logs.Next() {
+		_ = r // TODO: print logs
+	}
+	if err = logs.Close(); err != nil && !errors.Is(err, io.EOF) && !errors.Is(err, context.Canceled) {
```

Comment:
Shouldn't you check if you at least got one log?

Reply:
I'm not following... What would you do if you don't receive any logs. Also, you might receive no logs b/c reading the first byte failed with an error. That error is available via close. (This is the same pattern as reading e.g. a file with a `Scanner`).

Reply:
I typically use a mocked HTTP client for unit tests. That prevents having to run the actual service (that would make it an integration test). Then the REST call would return a fixed set of data. You would also test if your protobuf decoding logic works, because you know what to expect and how the decoded output should look like.

There are multiple ways to accomplish this, see https://stackoverflow.com/a/51227660/956435, 
https://medium.com/zus-health/mocking-outbound-http-requests-in-go-youre-probably-doing-it-wrong-60373a38d2aa or just substitute the HTTP client's transport layer.

Of course, the unit test won't be able to detect problems that arise from changes made to the server-side. You'll need an integration test to check that.

Reply:
This is an example on how to use the API call. I integration tests are done at the server. (Ref [here](https://github.com/miniohq/minkms/blob/main/server_test.go#L153) - as an example).

I agree that having runnable examples for the API calls would be nice. However, I would prefer to add this separately as a comprehensive solution for all API calls - instead of one-offs. Otherwise, we might end up with different special cases. However, I will take a look at this 👍